### ISSUE TITLE
Use SVG icons instead of bitmaps

### ISSUE THREAD

**Issue by jonkoops** [Leaflet/Leaflet#8756]
Now that all browsers we're targeting support SVG we should make an effort to replace the bitmap icons we have. I am marking this as 'help wanted' since someone will have to recreate the icons.

**Comment by jonkoops** [comment]
@mourner would you happen to have the original assets for the icons? Perhaps we can reach out on our socials to see if there is someone out there who'd donate some time to convert/redesign the icons we have.

**Comment by IvanSanchez** [comment]
> would you happen to have the original assets for the icons? 

@jonkoops You mean https://github.com/Leaflet/Leaflet/tree/main/src/images ?

**Comment by jonkoops** [comment]
@IvanSanchez No, I am referring to the ones in the [`dist`](https://github.com/Leaflet/Leaflet/tree/9f35179419aad62d5d735f49a70e7a79fd67cfd5/dist/images) directory. If we replace these we can get rid of a lot of 'retina' based code paths for the icons.

**Comment by mourner** [comment]
> @mourner would you happen to have the original assets for the icons?

@jonkoops I'm confused, aren't they the ones in the `src/images` dir? The only SVG that's missing is for the marker shadow, easy enough to recreate.

**Comment by jonkoops** [comment]
Ahhhh, yes. My mistake, I was under the impression that those were different icons. Cool, we can mostly use those then.

**Comment by jonkoops** [comment]
Created a draft PR under #8766 so we can collaborate on this.
